File: Cadastro_de_alunos/app/src/main/java/com/example/brendon/registrodealunosv1/Adapters/AlunoAdapter.java
package com.example.brendon.registrodealunosv1.Adapters;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.design.widget.Snackbar;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.PopupMenu;
import android.widget.TextView;
import android.widget.Toast;

import com.example.brendon.registrodealunosv1.Formularios.FormularioAddAluno;
import com.example.brendon.registrodealunosv1.Models.Aluno;
import com.example.brendon.registrodealunosv1.R;

import io.objectbox.Box;

public class AlunoAdapter extends RecyclerView.Adapter<AlunoAdapter.AlunoViewHolder> {

    public static String ID = "idAluno";

    private Context context;
    private Box<Aluno> listAlunos;


    public AlunoAdapter(Context context, Box<Aluno> listAlunos) {
        this.listAlunos = listAlunos;
        this.context = context;
    }

    @NonNull
    @Override
    public AlunoViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.card_aluno, viewGroup, false);
        return new AlunoViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull AlunoViewHolder holder, final int position) {
        final Aluno alunoAtual = this.listAlunos.getAll().get(position);

        holder.txtNome.setText("Nome: " + alunoAtual.getNome());
        holder.txtCurso.setText("Curso: " + alunoAtual.getCurso());
        holder.txtFaculdade.setText("Faculdade " + alunoAtual.getFaculdade());

        imageMenus(holder.itemView, alunoAtual, position);

        popMenu(holder.itemView, alunoAtual, position);
    }

    private void popMenu(View itemView, Aluno aluno, int position){
        itemView.setOnLongClickListener((view ) -> {
            PopupMenu popup = new PopupMenu(context, view);
            popup.getMenuInflater().inflate(R.menu.pop_menu, popup.getMenu());

            popup.setOnMenuItemClickListener((item) -> {

                switch (item.getItemId()){
                    case R.id.op_editar:

                        Intent intent = new Intent(context, FormularioAddAluno.class);
                        intent.putExtra(ID, aluno.id);
                        context.startActivity(intent);
                        notifyItemChanged(position);
                        break;

                    case R.id.op_remover:

                        AlertDialog alerta;
                        AlertDialog.Builder builder = new AlertDialog.Builder(context);
                        builder.setTitle("Remover");
                        builder.setMessage("Deseja remover aluno(a): " + aluno.getNome()+" ?");
                        builder.setPositiveButton("Sim", (arg0, arg1) -> {
                            listAlunos.remove(aluno);
                            notifyItemRemoved(position);
                            Snackbar.make(view,"Removido com sucesso!", Snackbar.LENGTH_LONG).show();
                        });
                        builder.setNegativeButton("Nao", (arg0, arg1) -> {
                        });
                        alerta = builder.create();
                        alerta.show();
                        break;
                }

                return false;
            });

            popup.show();

            return true;
        });
    }

    private void imageMenus(View itemView, Aluno aluno, int position){
        ImageButton excluir = itemView.findViewById(R.id.excluir_aluno);
        ImageButton editar = itemView.findViewById(R.id.editar_aluno);

        editar.setOnClickListener(l ->{
            Intent intent = new Intent(context, FormularioAddAluno.class);
            intent.putExtra(ID, aluno.id);
            context.startActivity(intent);
            notifyItemChanged(position);

        });

        editar.setOnLongClickListener(l->{
            Toast.makeText(context, "Editar", Toast.LENGTH_LONG).show();
            return true;
        });

        excluir.setOnLongClickListener(l ->{
            Toast.makeText(context, "Excluir", Toast.LENGTH_LONG).show();
            return true;
        });

        excluir.setOnClickListener(l ->{
            AlertDialog alerta;
            AlertDialog.Builder builder = new AlertDialog.Builder(context);
            builder.setTitle("Remover");
            builder.setMessage("Deseja remover aluno(a): " + aluno.getNome()+" ?");
            builder.setPositiveButton("Sim", (arg0, arg1) -> {
                listAlunos.remove(aluno);
                notifyItemRemoved(position);
                Snackbar.make(l,"Removido com sucesso!", Snackbar.LENGTH_LONG).show();
            });
            builder.setNegativeButton("Nao", (arg0, arg1) -> {
            });
            alerta = builder.create();
            alerta.show();
        });
    }

    @Override
    public int getItemCount() {
        return this.listAlunos.getAll().size();
    }

    public class AlunoViewHolder extends RecyclerView.ViewHolder {
        TextView txtNome, txtCurso, txtFaculdade;

        public AlunoViewHolder(View view) {
            super(view);
            txtNome = view.findViewById(R.id.view_nome_aluno);
            txtCurso = view.findViewById(R.id.view_curso);
            txtFaculdade = view.findViewById(R.id.view_faculdade);

        }
    }
}

File: RegistroDeAlunosKotlinEJava/app/src/main/java/com/example/artur/registrodealunos/Adapter/AlunoAdapter.kt
package com.example.artur.registrodealunos.Adapter

import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.PopupMenu
import android.widget.Toast
import com.example.artur.registrodealunos.FormularioAlunoActivity
import com.example.artur.registrodealunos.Modelo.Aluno
import com.example.artur.registrodealunos.R
import io.objectbox.Box
import kotlinx.android.synthetic.main.item_aluno.view.*

class AlunoAdapter (private val context: Context,
                    private val alunosBox: Box<Aluno>): RecyclerView.Adapter<AlunoAdapter.ViewHolder>(){

    companion object {
        const val ID = "id"

    }

    class ViewHolder(itemView: View): RecyclerView.ViewHolder(itemView){
        fun bind(aluno: Aluno){
            val nome = itemView.text_nome
            val curso = itemView.text_curso

            nome.text = aluno.nome
            curso.text = aluno.curso
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_aluno, parent,false))
    }

    override fun getItemCount(): Int {
        return alunosBox.all.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val aluno = this.alunosBox.all[position]
        holder.bind(aluno)

        popMenu(holder.itemView, aluno, position)
    }

    private fun popMenu(itemView: View, aluno: Aluno, position: Int){

        itemView.setOnLongClickListener { view ->
            val popup = PopupMenu(context, view)
            popup.menuInflater.inflate(com.example.artur.registrodealunos.R.menu.pop_menu, popup.menu)
            popup.setOnMenuItemClickListener { item ->
                when (item.itemId){
                    R.id.op_editar -> editar(aluno, position)
                    R.id.op_remover -> remover(aluno, position)

                }
                false
            }
            popup.show()

            true
        }
    }

    private fun editar(aluno: Aluno, position: Int){
        val intent = Intent(context, FormularioAlunoActivity::class.java)
        intent.putExtra(ID, aluno.id)
        context.startActivity(intent)
        notifyItemChanged(position) //atualiza a lista
    }

    private fun remover(aluno: Aluno, position: Int){
        val alertDialog = AlertDialog.Builder(context)

        alertDialog.setTitle("Remover")
            .setMessage("Deseja realmente remover ${aluno.nome}?")
            .setPositiveButton("SIM"){ _, _ ->
                this.alunosBox.remove(aluno)
                notifyItemRemoved(position) //atualiza a lista
                notifyItemRangeChanged(position, itemCount)
                Toast.makeText(context, "${aluno.nome} apagado", Toast.LENGTH_SHORT).show()
            }
            .setNegativeButton("NÃO"){_, _ ->}
            .create()
            .show()
    }
}